xdsl/dialects/memref.py

```diff
@@ -183,10 +183,15 @@ class Alloca(Operation):
     @staticmethod
     def get(return_type: Attribute,
             alignment: int,
-            shape: Optional[List[int | AnyIntegerAttr]] = None) -> Alloca:
+            shape: Optional[List[int | AnyIntegerAttr]] = None,
```

Comment:
How about this?

```suggestion
            shape: Optional[List[int | AnyIntegerAttr]] = [1],
```

And later `shape = list(shape)`. Making it Optional seems a bit ugly, and if we're good about making copies then it seems like the list copy is not a big penalty.

Reply:
I don't really like this. I think this is too easy to break, by removing an "unnecessary" `list(...)` cast... We should stick with the default of `None + if`.

Reply:
The problem is this:
![image](https://user-images.githubusercontent.com/14147713/220633664-5bf67703-7d66-4af4-a31b-41670e54fab3.png)

Here the second `test` function uses the fix @superlopuh proposed. Now if you look at the function body, we see a `list` cast, that does "nothing" (in fact `assert output == input` will hold!). But if this supposed nop is removed in a future commit, it might produce errors that are difficult to debug.

The `None + if` approach is a widely accepted practice in pyhton and we should stick with that imo.

Reply:
I'm fine for both options, but:
* If we take the `list(...)` approach, we should add a `#copy` comment
* If we take the `= None` approach, we should probably document the default value in the function.

Reply:
I am wondering why we even have a default value of `[1]` here. The mlir one doesn't have a default size afaik: https://mlir.llvm.org/docs/Dialects/MemRef/#memrefalloca-mlirmemrefallocaop